source/common/runtime/runtime_features.cc

```diff
@@ -60,6 +60,7 @@ constexpr const char* runtime_features[] = {
     "envoy.reloadable_features.strict_authority_validation",
     "envoy.reloadable_features.reject_unsupported_transfer_encodings",
     // Begin alphabetically sorted section.
+    "envoy.api.enable_deprecated_warning",
```

Comment:
Sorry I should have commented on this before but 2 things:
1) I think the name of this should be a bit more specific to what it is guarding.
2) This needs a release note.

Reply:
> I think the name of this should be a bit more specific to what it is guarding.

Any suggestions? The current name was suggested by @htuch...

Reply:
`envoy.api.enable_deprecated_v2_api_warning` ?